src/Features/Core/Portable/InlineHints/InlineHint.cs

```diff
@@ -16,18 +16,21 @@ namespace Microsoft.CodeAnalysis.InlineHints
         public readonly TextSpan Span;
         public readonly ImmutableArray<TaggedText> DisplayParts;
         private readonly Func<Document, CancellationToken, Task<ImmutableArray<TaggedText>>>? _getDescriptionAsync;
+        private readonly Func<string>? _getReplacementText;
 
         public InlineHint(
             TextSpan span,
             ImmutableArray<TaggedText> displayParts,
-            Func<Document, CancellationToken, Task<ImmutableArray<TaggedText>>>? getDescriptionAsync = null)
+            Func<Document, CancellationToken, Task<ImmutableArray<TaggedText>>>? getDescriptionAsync = null,
+            Func<string>? getReplacementText = null)
```

Comment:
this changes an API that our partners can call.  add a new overload but keep the existing constructor around for binary compat.

Reply:
instead of a Func<string>, have this be either a string, or a TextChangeRange if we want more flexibility on where to insert this.